crates/subspace-service/src/lib.rs

```diff
@@ -446,7 +450,12 @@ where
         telemetry: telemetry.as_mut(),
     })?;
 
-    network_starter.start_network();
+    let network_starter = if run_start_network {
```

Comment:
What about returning it unconditionally and starting explicitly even in `subspace-node`? I find this new boolean flag awkward.